### DISCUSSION TITLE
How to make a rule both silent and atomic?

### DISCUSSION THREAD

**Discussion by hillin** [pest-parser/pest#679] [Q&A]
I'd like a rule to be silent, but also atomic so it cannot have implicit whitespaces.
Some attempts:

```pest
// does not compile
foo_and_bar = _@{ foo ~ bar }

// does not compile
foo_and_bar = @_{ foo ~ bar }

// does not work, it hides `_foo_and_bar` but not `foo_and_bar`
foo_and_bar = @{ foo ~ bar }
_foo_and_bar = _{ foo_and_bar }
```

**Comment by Alextopher** [comment]
Being atomic means all the children rules are silent. If the atomic was also silent that would make the whole chain silent. Maybe a bit more details about exactly what you plan to use this for would help. You probably want the type above `_foo_and_bar` to be silent.

Such as:
```
foo_and_bar = @{ foo ~ bar }
_foo_and_bar = _{ foo_and_bar }
list = _{ SOI ~ _foo_and_bar* ~ EOI }
```

This gives you `[ foo_and_bar [], foo_and_bar [], foo_and_bar [], ... ]` as the final Pairs object

**Reply by hillin** [reply]
I have a certain pattern repeatedly appear in various places, so I'd like to abstract it out and made a rule (`foo_and_bar`). The thing is, I don't want to have `foo_and_bar` in the output tokens (only using it as an alias), so it'd better be silent. This all works great until I want to manually control the whitespaces in the pattern, which requires `foo_and_bar` to be atomic (or actually, composite atomic in my case).

**Reply by Alextopher** [reply]
Ah so you actually want the whole chain silent. I cannot seem to get any pattern to work expect putting `foo_and_bar` into the `COMMENT` or `WHITESPACE` block. Not sure if anything else can be done until there is an atomic silent specifier is created
